src/utils/promote.ts

```diff
@@ -299,29 +299,38 @@ async function getOrSelectApplication(
 		$expand: {
 			is_for__device_type: { $select: ['slug', 'is_of__cpu_architecture'] },
 		},
-	} satisfies BalenaSdk.PineOptions<BalenaSdk.Application>;
+	} satisfies BalenaSdk.Pine.ODataOptionsWithoutCount<
+		BalenaSdk.Application['Read']
+	>;
```

Comment:
Why `satisifies` instead of `options:  BalenaSdk.Pine.ODataOptionsWithoutCount<BalenaSdk.Application['Read']>`?

Reply:
> They are equivalent so I don't have any strong preference

The difference would be that we don't have to cast everywhere

Reply:
in fact, the best way here is to use `as const` simply so that the strong inference can infer the follow up types :+1:

Reply:
I don't see the follow up types getting inferred, the call to `await sdk.models.application.getAllDirectlyAccessible(options);` below still expects `options` to only be the type from when we initialized it

Reply:
Yes @myarmolinsky and for that call to have strong typed response (which we want, as this variable is used after on the filter) we need the typing to be a const readonly type (as const) rather than simply satisfying ODataOptionsWithoutCount. The `options as...` in the middle is just because we are adding $filters on the fly and that won't work with const (but that is fine as $filter does not currently affect the response type)